libs/sdk-ui-ext/src/internal/utils/embeddingCodeGenerator/getWebComponentsCodeGenerator.ts

```diff
@@ -21,14 +21,20 @@ export const getWebComponentsCodeGenerator = (
     insight: IInsight,
     codeOption: IWebComponentsOptions,
 ) => {
-    const { displayTitle, customTitle, customHeight, height, locale, allowLocale } = codeOption;
-    return WEB_COMPONENTS_EMBED_CODE.replace(HOST_NAME, window.location.hostname)
-        .replace(WORKSPACE_ID, workspaceId)
-        .replace(INSIGHT_ID, insight.insight.identifier)
-        .replace(
-            INSIGHT_TITLE,
-            displayTitle ? `\n    title="${customTitle ? CUSTOM_TITLE : insight.insight.title}"` : "",
-        )
-        .replace(LOCALE, allowLocale ? `\n    locale="${locale}"` : "")
-        .replace(CUSTOM_HEIGHT, customHeight ? `\n    style="${height}"` : "");
+    const { displayTitle, customTitle, customHeight, height, locale, allowLocale, unit } = codeOption;
+    return (
+        WEB_COMPONENTS_EMBED_CODE.replace(HOST_NAME, window.location.hostname)
+            .replace(WORKSPACE_ID, workspaceId)
+            .replace(INSIGHT_ID, insight.insight.identifier)
+            .replace(
+                INSIGHT_TITLE,
+                displayTitle ? `\n    title="${customTitle ? CUSTOM_TITLE : insight.insight.title}"` : "",
+            )
+            .replace(LOCALE, allowLocale ? `\n    locale="${locale}"` : "")
+            // when the unit is px, the height is a number, so we need to add the unit
+            .replace(
+                CUSTOM_HEIGHT,
+                customHeight ? `\n    style="height: ${height}${unit === "px" ? "px" : ""}"` : "",
```

Comment:
Just a confirmation: can `unit` be something like `%`?  If yes, should we add `%` right after `${height}`?

Reply:
I have updated the code. We should add the `px` when the height is a number. Otherwise, the height already has its unit.